android/java/org/chromium/chrome/browser/crypto_wallet/presenters/SolanaInstructionPresenter.java

```diff
@@ -128,18 +140,105 @@ public List<TwoLineItemText> accountParamDataToList() {
         return twoLineItemDataSources;
     }
 
+    public List<TwoLineItemText> programIdToList() {
+        return Arrays.asList(new TwoLineItemText(null, mSolanaInstruction.programId));
+    }
+
     public List<TwoLineItemText> dataToList() {
         return Arrays.asList(new TwoLineItemText(null, Arrays.toString(mSolanaInstruction.data)));
     }
 
+    // Get lamport from decoded data params
+    public String getLamportAmount() {
+        if (isDecodedDataPresent && mSolanaInstruction.decodedData.params != null) {
```

Comment:
couldn't `mSolanaInstruction` be null? Shouldn't we do a check on that? I think it can as a default constructor `SolanaInstructionPresenter` is accessible.

Reply:
the above is related to all usages of `mSolanaInstruction` thought the class

Reply:
I have an assert here for [null check](https://github.com/brave/brave-core/blob/b10d98c011cb6640447589b1a143b90cabd04732/android/java/org/chromium/chrome/browser/crypto_wallet/presenters/SolanaInstructionPresenter.java#L39).
There is no default constructor because java doesn't provide one if we define a parameterised constructor.